miner/block_builder.go

```diff
@@ -180,9 +182,15 @@ func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.AtxId, eligibil
 	} else if id == config.Genesis+1 {
 		votes = append(votes, config.GenesisId)
 	} else {
-		votes, err = t.hareResult.GetResult(id - 1)
-		if err != nil {
-			return nil, errors.New(fmt.Sprintf("didn't receive hare result for layer %v %v", id-1, err))
+
+		bottom := types.LayerID(1)
+		if id > t.hdist {
+			bottom = id - t.hdist
+		}
+
+		votes, err = t.hareResult.GetResult(bottom, id-1)
+		if len(votes) == 0 {
```

Comment:
votes can be nil, what is the expected behaviour then? I'd recommend checking err first

Reply:
yes, thanks did that before i added the err